packages/route-viewer-overlay/src/index.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect } from "react";
+import { Stop } from "@opentripplanner/types";
+import { Source, Layer, useMap, LngLatLike } from "react-map-gl";
+import { LngLatBounds } from "maplibre-gl";
+
+import polyline from "@mapbox/polyline";
+import pointInPolygon from "point-in-polygon";
+
+// helper fn to check if geometry has been populated for all patterns in route
+const isGeomComplete = (routeData: {
+  patterns: { id: string; geometry?: { points: [number, number][] } }[];
+}) => {
+  return (
+    routeData &&
+    routeData.patterns &&
+    Object.values(routeData.patterns).every(
+      ptn => typeof ptn?.geometry !== "undefined"
+    )
+  );
+};
+
+/**
+ * helper function that removes all points from array of points that are
+ * within flex zones defined in an array of stops
+ * @param {*} stops   OTP stops response
+ * @param {*} points  Array of coordinates to clip
+ * @returns           The array of coordinates without coordinates within the stops
+ */
+const removePointsInFlexZone = (stops: Stop[], points: [number, number][]) => {
+  // First, go through all stops to find flex zones
+  const bboxes =
+    stops
+      ?.map(stop => {
+        if (stop.geometries?.geoJson?.type !== "Polygon") {
+          return null;
+        }
+        return stop.geometries.geoJson.coordinates?.[0] || null;
```

Comment:
Wouldn't this result in more passes over the array? Normally I'd make the sacrifice for readability but this method gets called very often and blocks render

Reply:
If anything, should we remove that second filter and merge it into the map?

Reply:
I'd separate the map and the filtering, unless there is some benchmark that justifies otherwise (in that case, add a comment with the results of such benchmark)